src/sentry_plugins/github/endpoints/webhook.py

```diff
@@ -273,9 +273,120 @@ def __call__(self, event, organization=None):
             self._handle(event, org, is_apps)
 
 
+class PullRequestEventWebhook(Webhook):
+    # https://developer.github.com/v3/activity/events/types/#pullrequestevent
+    def __call__(self, event, organization):
+        action = event['action']
+        is_apps = 'installation' in event
+
+        try:
+            repo = Repository.objects.get(
+                organization_id=organization.id,
+                provider='github_apps' if is_apps else 'github',
+                external_id=six.text_type(event['repository']['id']),
+            )
+
+        except Repository.DoesNotExist:
+            raise Http404()
+
+        # We need to track GitHub's "full_name" which is the repository slug.
+        # This is needed to access the API since `external_id` isn't sufficient.
+        if repo.config.get('name') != event['repository']['full_name']:
+            repo.config['name'] = event['repository']['full_name']
+            repo.save()
+
+        if action == 'opened':
+            self._handle_created(event, organization, repo, is_apps)
+        if action == 'edited':
+            self._handle_updated(event, organization, repo)
+        if action == 'synchronize':
+            self._handle_sync(event, organization, repo)
+
+    def _handle_created(self, event, organization, repo, is_apps):
+        """PR was created"""
+
+        client = GitHubClient()
```

Comment:
i think you're going to need to pass auth into this for any private repos (and we'd want to for public ones due to api rate limits). since this is a webhook and there isn't a user making the request (so it's unclear which auth we should use) we're probably going to have to think of a work around.

Reply:
we do the same thing in commit i think, why does it work there? could we store auth on the repo?

Reply:
that's just the user endpoint, which doesn't require auth: https://github.com/getsentry/sentry-plugins/blob/master/src/sentry_plugins/github/endpoints/webhook.py#L157

Reply:
dang